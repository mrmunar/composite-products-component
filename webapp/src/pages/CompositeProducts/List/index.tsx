import React, { useEffect } from 'react';
import { useDispatch } from 'react-redux'

import { getProducts } from '../../../redux/actions/products';

const List = () => {
    const dispatch = useDispatch();

    //get composite products on render
    useEffect(() => {
        dispatch(getProducts());
    });

    return (
        // title
        // add button
        // list
            <ul className="list-group text-center">
                <li className="list-group-item">Cras justo odio</li>
                <li className="list-group-item">Dapibus ac facilisis in</li>
                <li className="list-group-item">Morbi leo risus</li>
                <li className="list-group-item">Porta ac consectetur ac</li>
                <li className="list-group-item">Vestibulum at eros</li>
            </ul>
    );
}

export default List;