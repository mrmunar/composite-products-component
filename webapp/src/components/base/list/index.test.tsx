import React from 'react';
import { render, cleanup, fireEvent, screen } from '@testing-library/react';
import List from './';

afterEach(cleanup);

const testData = [
    { id: 1, name: 'item1' },
    { id: 2, name: 'item2' },
    { id: 3, name: 'item3' },
];

test('should take a snapshot', () => {
    const { baseElement } = render(<List items={testData} />);
    expect(baseElement).toMatchSnapshot();
});

test('should render items', () => {
    const { getByTestId } = render(<List items={testData} />);
    expect(getByTestId('list')).toHaveTextContent('item1');
    expect(getByTestId('list')).toHaveTextContent('item2');
    expect(getByTestId('list')).toHaveTextContent('item3');
});


