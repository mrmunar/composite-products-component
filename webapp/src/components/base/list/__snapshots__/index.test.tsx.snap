// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<body>
  <div>
    <ul
      class="list-group"
      data-testid="list"
    >
      <li
        class="list-group-item"
        data-testid="list-item"
      >
        item1
      </li>
      <li
        class="list-group-item"
        data-testid="list-item"
      >
        item2
      </li>
      <li
        class="list-group-item"
        data-testid="list-item"
      >
        item3
      </li>
    </ul>
  </div>
</body>
`;
